includes/Endpoints/DeletePost.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace VPlugins\SMPostConnector\Endpoints;
+
+use WP_REST_Request;
+use WP_REST_Response;
+use VPlugins\SMPostConnector\Middleware\AuthMiddleware;
+
+class DeletePost {
+    protected $auth_middleware;
+
+    public function __construct() {
+        $this->auth_middleware = new AuthMiddleware();
+        add_action('rest_api_init', [$this, 'register_routes']);
+    }
+
+    public function register_routes() {
+        register_rest_route('sm-connect/v1', '/delete-post', [
+            'methods' => 'DELETE',
+            'callback' => [$this, 'delete_post'],
+            'permission_callback' => [$this->auth_middleware, 'permissions_check']
+        ]);
+    }
+
+    public function delete_post(WP_REST_Request $request) {
+        // Get post ID and trash flag from request
+        $post_id = $request->get_param('id');
+        $trash = $request->get_param('trash');
+
+        if (!$post_id) {
+            return new WP_REST_Response([
+                'status' => 400,
+                'message' => 'Post ID is required'
+            ], 400);
+        }
+
+        // Check if post exists
+        if (get_post($post_id) === null) {
```

Comment:
Use [get_posts()](https://developer.wordpress.org/reference/functions/get_posts/) function instead of get_post(). So that we can enhance multi-delete in the future